### ISSUE TITLE
Provide SDL2 button mappings for convenience

### ISSUE THREAD

**Issue by emilyst** [emilyst/hid-nx-dkms#11]
I'm opening this issue to split off SDL2 mapping discussion from issue #7. Please refer to that issue for context.

**Comment by offalynne** [comment]
Correct. All that's required for support are additional mappings using hid-nx, since the GUID does come out uniquely. I have already committed maps for SNES and N64 pads.

**Comment by emilyst** [comment]
> Correct. All that's required for support are mapping strings composed while using hid-nx, since the GUID does come out uniquely. I have already committed maps for SNES and N64 pads.

Just checking, the GUID you refer to, what does that look like? (If it's based on the vendor and device ID, those actually weirdly change and even overlap depending on the situation.)

**Comment by nfp0** [comment]
Ah yes. I understand what you mean now @offalynne

**Comment by offalynne** [comment]
@emilyst see 
https://github.com/gabomdq/SDL_GameControllerDB/commit/67fdcddb33e37bf6234591449b2bc4aeb016b0d8
and
https://github.com/gabomdq/SDL_GameControllerDB/commit/dbcb77c22d554d34e4cec71919d86fbd645269b0

Mappings authored using SDL seem to provide a unique GUID with hid-nx (the "revision" field in the descriptor, I believe),
so no collision issue with the different mapping.

**Comment by emilyst** [comment]
> @emilyst see [gabomdq/SDL_GameControllerDB@67fdcdd](https://github.com/gabomdq/SDL_GameControllerDB/commit/67fdcddb33e37bf6234591449b2bc4aeb016b0d8) and [gabomdq/SDL_GameControllerDB@dbcb77c](https://github.com/gabomdq/SDL_GameControllerDB/commit/dbcb77c22d554d34e4cec71919d86fbd645269b0)
> 
> Mappings authored using SDL seem to provide a unique GUID with hid-nx (the "revision" field in the descriptor, I believe), so no collision issue with the different mapping.

Okay, cool. For what it's worth, SDL seems to call these "controller GUIDs," but they're not really GUIDs as the broader industry has defined them, but instead derived from the product, vendor, and name of the device.

Anywho, I'll try to remember to do this sometime soon.

> > If you want to open a dedicated issue, please do. If you have a new topic to discuss, even if it's related to an existing one, it's encouraged to open a new issue for that topic. That way it's easier for myself and others to keep track of.
> 
> For what it's worth, I am sympathetic to this and hope continuing the conversation in this new issue (thanks for making it) is suitable to you. I apologize for sidebarring in the other issue. I have learned the hard way through years of managing large repos with many collaborators to try and reach people immediately wherever they are actively posting on any given subject, even where tangential, because attention is scarce and people are busy (eg. your dialogue with the upstream submitter Daniel). I realize this can muddy issue organization for a smaller project.

Thanks for being considerate. My ability to keep on top of and respond to issues depends on the issues being focused and well organized, so I suppose I'm unusual in that respect, but it really helps me.

**Comment by offalynne** [comment]
> SDL seems to call these "controller GUIDs," but they're not really GUIDs as the broader industry has defined them

Correct, they're identifier strings with a number of different formats and features meant to be unique within the list (with several exceptions) and don't meet the common definition of GUID
